### ISSUE TITLE
More than 15 pdfs causes an invalid file

### ISSUE THREAD

**Issue by happilymarrieddad** [wubzz/pdf-merge#8]
Everytime I try to generate a pdf from more than 15 other pdfs, it returns an invalid pdf file that none of the readers I have can parse. It's almost like there is a memory issue or something.

**Comment by wubzz** [comment]
@happilymarrieddad Going to need some more information on this, such as PDF filenames and filesizes, what your node process CPU/Memory usage is.

**Comment by happilymarrieddad** [comment]
oh sorry man,

![selection_130](https://cloud.githubusercontent.com/assets/6416323/18246764/0903e56e-732b-11e6-9788-77ca4250dd68.png)

"/var/www/fusion/public/images/temp/printlabel-116251044535424852258676.pdf" "/var/www/fusion/public/images/temp/printlabel-116252042333645460811575.pdf" "/var/www/fusion/public/images/temp/printlabel-11625305474470039689505.pdf" "/var/www/fusion/public/images/temp/printlabel-11625408987722079016849.pdf" "/var/www/fusion/public/images/temp/printlabel-11625505457466533471083.pdf" "/var/www/fusion/public/images/temp/printlabel-11625605714969587011638.pdf" "/var/www/fusion/public/images/temp/printlabel-116257021536666893112044.pdf" "/var/www/fusion/public/images/temp/printlabel-1162580028078796213059976.pdf" "/var/www/fusion/public/images/temp/printlabel-1162590690549601558802.pdf" "/var/www/fusion/public/images/temp/printlabel-11626006436126211579158.pdf" "/var/www/fusion/public/images/temp/printlabel-11626108978347740286463.pdf" "/var/www/fusion/public/images/temp/printlabel-116262010192026265854248.pdf" "/var/www/fusion/public/images/temp/printmultiplelabels-05170397100087565.pdf"

more like that... 
pdf-merge doesn't give me any errors but here is what easy-pdf-merge gives me.

![selection_131](https://cloud.githubusercontent.com/assets/6416323/18246802/4a3bdf32-732b-11e6-94fa-17328463662b.png)

I understand you're not using java but I can't seem to get an error with your package. It just returns an invalid pdf. I open it in any viewer and it fails. Thanks!

**Comment by happilymarrieddad** [comment]
I can send you the pdf's we're trying to merge.. it's not sensitive material.

**Comment by happilymarrieddad** [comment]
`Receivables.mergePdfs = function(data,respond) {
	var results = {rawtimestamp:data.rawtimestamp};
	var pdfs = data.pdfs;
	var filepath = '';

	function finish(){
		respond(null,filepath);
	}

	if (process.env.ENV == 'production') {
		filepath = "/var/www/fusion/public/images/temp/printmultiplelabels-"+data.rawtimestamp+".pdf";
	} else {
		filepath = __dirname + "/../public/images/temp/printmultiplelabels-"+data.rawtimestamp+".pdf";
	}

	for (var i = 0; i < pdfs.length; i++) {
		if (process.env.ENV == 'production'){
			pdfs[i] = "/var/www/fusion/public" + pdfs[i];
		} else {
			pdfs[i] = __dirname + "/../public" + pdfs[i];
		}
	}

	async.series([
		function(callback) {
			var pdfMerge = new PDFMerge(pdfs);
			pdfMerge.asNewFile(filepath).merge(function(error, filePath) {
				results.filePath = filepath;
				if (process.env.ENV == 'production') {
				  	Services.sendToApplicationServers(filepath,'/public/images/printmultiplelabels-'+data.rawtimestamp+'.pdf',function(err) {
				  		if (err) { return respond(err) }
					  	return callback()
				  	})
				} else {
					return callback();
				}
			});
		}
	],finish)
}`

**Comment by prescottprue** [comment]
Any movement on this? I am not receiving any errors either.

**Comment by wubzz** [comment]
@happilymarrieddad @prescottprue Generally this module will only throw an error if whatever the thirdparty module **PDFtk** spits back out is stderr.. I tested a merge of 50 PDF documents just now and it worked completely fine, all with very long names like you suggested.

My guess is this has something to do with the actual PDF files. Password protected? Encrypted? Corrupt? Hard to guess.. If you don't mind sharing them somewhere, I'll have a closer look.

My first through maybe it was due to duplicate names, since you listed a couple of duplicates, but doesn't look like that's the case either (tested, works)
